src/rest.cpp

```diff
@@ -107,6 +107,23 @@ static CTxMemPool* GetMemPool(const std::any& context, HTTPRequest* req)
     return node_context->mempool.get();
 }
 
+/**
+ * Get the node context chainstatemanager.
+ *
+ * @param[in]  req The HTTP request, whose status code will be set if node
+ *                 context chainstatemanager is not found.
+ * @returns        Pointer to the chainstatemanager or nullptr if none found.
+ */
+static ChainstateManager* GetChainman(const std::any& context, HTTPRequest* req)
+{
+    auto node_context = util::AnyPtr<NodeContext>(context);
+    if (!node_context) {
+        RESTERR(req, HTTP_NOT_FOUND, "Chainman disabled or instance not found");
```

Comment:
In commit "rest: Add GetChainman function and use it" (a839078c20f4ac3d92e14af7b287f2f0fddbc61e)

If this condition is never expected, then maybe this should use `HTTP_INTERNAL_SERVER_ERROR` to flag that this is a bug in request handling, not an invalid request.